experiment/commenter/main.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Commenter provides a way to --query for issues and append a --comment to matches.
+//
+// The --user and --token determines who interacts with github.
+// By default commenter runs in dry mode, add --wet to make it leave comments.
+// The --updated, --include-closed, --ceiling options provide minor safeguards
+// around leaving excessive comments.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/test-infra/prow/github"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.query, "query", "", "See https://help.github.com/articles/searching-issues-and-pull-requests/")
+	flag.DurationVar(&o.updated, "updated", 2*time.Hour, "Filter to issues unmodified for at least this long if set")
+	flag.BoolVar(&o.includeClosed, "include-closed", false, "Match closed issues if set")
+	flag.BoolVar(&o.wet, "wet", false, "Mutate github if set")
+	flag.StringVar(&o.comment, "comment", "", "Append the following comment to matching issues")
+	flag.IntVar(&o.ceiling, "ceiling", 3, "Maximum number of issues to modify, 0 for infinite")
+	flag.StringVar(&o.user, "user", "", "Github username of commenter")
+	flag.StringVar(&o.token, "token", "", "Path to github token")
+	flag.Parse()
+	return o
+}
+
+func parseHTMLURL(url string) (string, string, int, error) {
+	// https://github.com/batterseapower/pinyin-toolkit/issues/132`)
+	re := regexp.MustCompile(`.+/(.+)/(.+)/(issues|pull)/(\d+)$`)
+	mat := re.FindStringSubmatch(url)
+	if mat == nil {
+		return "", "", 0, fmt.Errorf("failed to parse: %s", url)
+	}
+	n, err := strconv.Atoi(mat[4])
+	if err != nil {
+		return "", "", 0, err
+	}
+	return mat[1], mat[2], n, nil
+}
+
+func makeQuery(query string, includeClosed bool, minUpdated time.Duration) string {
+	parts := []string{query}
+	if !includeClosed {
+		parts = append(parts, "is:open")
+	}
+	if minUpdated != 0 {
+		latest := time.Now().Add(-minUpdated)
+		parts = append(parts, "updated:<="+latest.Format(time.RFC3339))
+	}
+	return strings.Join(parts, " ")
+}
+
+type options struct {
```

Comment:
Minor comment, but why use the `options` struct instead of just something like
```go
var (
	flag.Bool("wet", false, "Mutate github if set")
	...
)
```
That's a much more common pattern for small binaries.